src/main/java/seedu/weme/model/ModelManager.java

```diff
@@ -185,6 +191,24 @@ public void commitMemeBook() {
         versionedMemeBook.commit();
     }
 
+    @Override
+    public void cleanMemeStorage() {
+        try {
+            Set<File> filesToKeep = new HashSet<>();
+            for (Meme meme : versionedMemeBook.getMemeList()) {
+                File file = meme.getFilePath().getFilePath().toFile();
+                filesToKeep.add(file);
+            }
+
+            Files.walk(getMemeImagePath())
```

Comment:
I think `Files.list` will do. `walk` will contain any subdirectory in the meme image path, if there are any. `File::delete` will fail in this case because it might be attempting to delete a directory that is not empty.